clang/include/clang/Basic/Attr.td

```diff
@@ -1023,6 +1023,26 @@ def SYCLKernel : InheritableAttr {
   let Documentation = [Undocumented];
 }
 
+def SYCLScope : Attr {
+  let Spellings = [GNU<"sycl_scope">];
+  let Args = [EnumArgument<"level", "Level", ["work_group", "work_item"],
+                                             ["WorkGroup", "WorkItem"]>];
+  let Subjects = SubjectList<[Function]>;
+  let LangOpts = [SYCLIsDevice];
```

Comment:
not permissible in the host?  Is that on purpose?

Reply:
Yes. This attribute makes sense only in the device compiler for now.